Source/player.h

```diff
@@ -454,6 +462,30 @@ struct PlayerStruct {
 		return blkper;
 	}
 
+	/**
+	 * @brief Return monster armor value after including player's armor piercing % (hellfire only)
+	 * @param monsterArmor - monster armor before applying % armor pierce
+	 * @param isMelee - indicates if it's melee or ranged combat
+	 */
+	int CalculateArmorPierce(int monsterArmor, bool isMelee) const
+	{
+		int tmac = monsterArmor;
+		if (gbIsHellfire && _pIEnAc > 0) {
+			int pIEnAc = _pIEnAc - 1;
+			if (pIEnAc > 0)
+				tmac >>= pIEnAc;
+			else
+				tmac -= tmac / 4;
+		}
+		if (isMelee && _pClass == HeroClass::Barbarian) {

```

Comment:
Can we check weapon to determine if its melee?

Reply:
what if someone fires an arrow and unequips the bow before the arrow hits? ;)